src/api/middleware/application.js

```diff
@@ -0,0 +1,19 @@
+import CompanyApplication, { CompanyApplicationRules } from "../../models/CompanyApplication.js";
+import { APIError, ErrorTypes } from "./errorHandler.js";
+import { StatusCodes as HTTPStatus } from "http-status-codes/build/cjs/status-codes.js";
+import { RECOVERY_LINK_EXPIRATION } from "../../models/constants/ApplicationStatus.js";
+
+export const exceededCreationTimeLimit = async (email) => {
+    const cursor = await CompanyApplication.findOne({ email, isVerified: false }).exec();
```

Comment:
```suggestion
    const application = await CompanyApplication.findOne({ email, isVerified: false }).exec();
```

Reply:
Isn't this use of `exec` wrong? Also, `findOne` should, when awaited, return the specified object, not a cursor, if i recall correctly.